skrub/tests/test_table_vectorizer.py

```diff
@@ -756,3 +758,148 @@ def test_changing_types_int_float():
+        cardinality_threshold=4,
+    )
+    table_vec_all_cols.fit(X)
+    for col in X.columns:
+        table_vec_one_col = TableVectorizer(
+            high_card_cat_transformer=GapEncoder(n_components=2, random_state=0),
+            cardinality_threshold=4,
+        )
+        table_vec_one_col.fit(X[[col]])
+        assert table_vec_one_col.get_feature_names_out() == [
+            feat
+            for feat in table_vec_all_cols.get_feature_names_out()
+            if feat.startswith(col)
+        ]
+
+
+@skip_if_no_parallel
+def test_parallelism() -> None:
+    # Test that parallelism works
+    X = _get_clean_dataframe()
+    # the gap encoder and the minhashencoder
+    # should be parallelized on all columns
+    # the one hot encoder should not be parallelized
+    for high_card_cat_transformer in [
+        lambda: GapEncoder(n_components=2, random_state=0),
+        OneHotEncoder,
+        lambda: MinHashEncoder(n_components=2),
+    ]:
+        table_vec_no_parallel = TableVectorizer(
+            high_card_cat_transformer=high_card_cat_transformer(),
+            cardinality_threshold=4,
+        )
+        X_trans = table_vec_no_parallel.fit_transform(X)
+        for joblib_backend in ["loky"]:
+            with joblib.parallel_backend(joblib_backend):
+                for n_jobs in [None, 2, -1]:
+                    table_vec = TableVectorizer(
+                        n_jobs=n_jobs,
+                        high_card_cat_transformer=high_card_cat_transformer(),
+                        cardinality_threshold=4,
+                    )
+                    X_trans_parallel = table_vec.fit_transform(X)
+                    assert_array_equal(X_trans, X_trans_parallel)
+                    assert table_vec.n_jobs == n_jobs
+                    # assert that all attributes are equal except for
+                    # the n_jobs attribute
+                    for attr in [
+                        "transformers_",
+                        "columns_",
+                        "types_",
+                        "imputed_columns_",
+                    ]:
+                        assert str(getattr(table_vec, attr)) == str(
+                            getattr(table_vec_no_parallel, attr)
+                        )
+                    # assert that get_feature_names_out gives the same result
+                    assert_array_equal(
+                        table_vec.get_feature_names_out(),
+                        table_vec_no_parallel.get_feature_names_out(),
+                    )
+                    # assert that get_params gives the same result expect for n_jobs
+                    # remove n_jobs from the dict
+                    params = table_vec.get_params()
+                    params.pop("n_jobs")
+                    params_no_parallel = table_vec_no_parallel.get_params()
+                    params_no_parallel.pop("n_jobs")
+                    assert str(params) == str(params_no_parallel)
```

Comment:
The problem is not the dictionnary, but the transformer values:
```
E                 Differing items:
E                 {'low_card_cat': OneHotEncoder(drop='if_binary', handle_unknown='infrequent_if_exist')} != {'low_card_cat': OneHotEncoder(drop='if_binary', handle_unknown='infrequent_if_exist')}
E                 {'high_card_cat': MinHashEncoder(n_components=2)} != {'high_card_cat': MinHashEncoder(n_components=2)}
```
Is there a cleaner whay do to this?

Reply:
Mmm, and it might also make it more reliable, since dictionaries can have different orders.

Reply:
@LilianBoulard, according to the linked thread, Pytest knows how to handle dictionaries, regardless of the order.
@LeoGrin, I see. Like Lilian said in another comment, maybe the best solution is to test elements individually. Otherwise, there is no solution apart from defining `__eq__` methods.

Reply:
@Vincent-Maladiere I think the simple `str` solution works well in this case, as sklearn puts the hyperparameters in the string representation. For the `transformers_` attribute, I've made a few functions to check equality, but they may be a bit overengineered. WDYT?

Reply:
Ok, let's keep it and move on! We can still correct it later.